sdk/src/auth.rs

```diff
@@ -139,6 +141,12 @@ impl ClientConfig {
         self
     }
 
+    /// Specify the user_agent header to be sent by the client.
+    pub fn with_user_agent(mut self, user_agent: String) -> Self {
```

Comment:
might do what we do above in `with_resolve`
```suggestion
    pub fn with_user_agent(mut self, user_agent: impl ToString) -> Self {
```

Reply:
Good call, added.